#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/humans/pedestrian/protos/Pedestrian.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/irobot/create/protos/Create.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/tables/protos/RoundTable.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/chairs/protos/Chair.proto"
EXTERNPROTO "../protos/TrackOne.proto"
EXTERNPROTO "../protos/LineTrack.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/BlanketFabric.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/apartment_structure/protos/Wall.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.34972221357804506 8.476486447158412e-07 -0.9368534428230244 3.141580026757764
  position 7.706616840168772 -0.8645008870121145 7.100248807657662
  follow "inside"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation -1.41 0 0
  name "inside"
  floorSize 13 11
  floorAppearance PBRAppearance {
    roughness 1
    roughnessMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/parquetry/chequered_parquetry_roughness.jpg"
      ]
    }
    metalness 0
    normalMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/parquetry/chequered_parquetry_normal.jpg"
      ]
    }
  }
}
RectangleArena {
  translation -1.41 -9.09 0
  name "outside"
  floorSize 13 7
  floorAppearance PBRAppearance {
    roughness 1
    roughnessMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/parquetry/chequered_parquetry_roughness.jpg"
      ]
    }
    metalness 0
    normalMap ImageTexture {
      url [
        "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/textures/parquetry/chequered_parquetry_normal.jpg"
      ]
    }
  }
}
RoundTable {
  translation 2.5 -2.55 -0.01560081417268655
  rotation -0.9347032785386005 0.31156775951285653 -0.17104184377318324 3.2725046332459707e-18
  name "round table BL"
  physics Physics {
  }
}
Chair {
  translation 1.68 -2.62 0
}
Chair {
  translation 2.45944 -3.38082 0
  rotation 0 0 1 1.57
  name "chair(4)"
}
Chair {
  translation 3.33078 -2.58215 0
  rotation 0 0 1 3.14
  name "chair(5)"
}
Chair {
  translation 2.44218 -1.69312 0
  rotation 0 0 1 -1.57
  name "chair(6)"
}
RoundTable {
  translation 2.5 2.5 -0.01560081417268655
  rotation -0.9347032785386005 0.31156775951285653 -0.17104184377318324 3.2725046332459707e-18
  name "round table BR"
  physics Physics {
  }
}
Chair {
  translation 2.43816 3.34688 0
  rotation 0 0 1 -1.57
  name "chair(9)"
}
Chair {
  translation 3.33907 2.63785 0
  rotation 0 0 1 3.14
  name "chair(8)"
}
Chair {
  translation 2.4635 1.63918 0
  rotation 0 0 1 1.57
  name "chair(7)"
}
Chair {
  translation 1.64 2.45 0
  name "chair(1)"
}
RoundTable {
  translation -2.5 2.5 -0.01560081417268655
  rotation -0.9347032785386005 0.31156775951285653 -0.17104184377318324 3.2725046332459707e-18
  name "round table TR"
  physics Physics {
  }
}
Chair {
  translation -2.69188 3.38285 0
  rotation 0 0 1 -1.57
  name "chair(12)"
}
Chair {
  translation -2.7165 1.6433 0
  rotation 0 0 1 1.57
  name "chair(11)"
}
Chair {
  translation -1.79093 2.64601 0
  rotation 0 0 1 3.14
  name "chair(10)"
}
Chair {
  translation -3.35 2.38 0
  name "chair(2)"
}
RoundTable {
  translation -2.5 -2.5 -0.01560081417268655
  rotation -0.9347032785386005 0.31156775951285653 -0.17104184377318324 3.2725046332459707e-18
  name "round table TL"
  physics Physics {
  }
}
Chair {
  translation -1.7492 -2.56399 0
  rotation 0 0 1 3.14
  name "chair(15)"
}
Chair {
  translation -2.42053 -3.34699 0
  rotation 0 0 1 1.57
  name "chair(14)"
}
Chair {
  translation -2.76786 -1.65738 0
  rotation 0 0 1 -1.57
  name "chair(13)"
}
Chair {
  translation -3.34 -2.54 0
  name "chair(3)"
}
Create {
  translation -5.41482223010862 0.6911363440295105 0.044595514917802094
  rotation -0.0004333885329366192 -0.0006816677368251451 -0.9999996737516849 0.9458959469339417
}
Pedestrian {
  translation -2.73 -7.55 1.27
  rotation 0 0 1 1.5708
}
Pedestrian {
  translation -3.47 -7.55 1.27
  rotation 0 0 1 1.5708
  name "pedestrian(1)"
  shirtColor 1 0 0
}
TrackOne {
  translation 0 -2.5 -0.0685
  rotation 1 0 0 1.57
  scale 0.06 0.07 0.07
}
TrackOne {
  translation 0 2.5 -0.0685
  rotation 1 0 0 1.57
  scale 0.06 0.07 0.07
}
DEF LineTrack Solid {
  translation 0.08 0 -0.049
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.168627 0.180392 0.192157
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.15 7.3 0.1
      }
    }
  ]
}
DEF LineTrack Solid {
  translation 2.44 0.01 -0.049
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.168627 0.180392 0.192157
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.15 1.05 0.1
      }
    }
  ]
  name "solid(5)"
}
DEF LineTrack Solid {
  translation 0.010009 -2.6 -0.049
  rotation 0 0 1 1.5708
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.168627 0.180392 0.192157
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.15 1.8 0.1
      }
    }
  ]
  name "solid(1)"
}
DEF LineTrack Solid {
  translation 0.00999026 2.56 -0.049
  rotation 0 0 1 1.5708
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.168627 0.180392 0.192157
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.15 1.8 0.1
      }
    }
  ]
  name "solid(2)"
}
DEF LineTrack Solid {
  translation -4.95001 2.59 -0.049
  rotation 0 0 1 1.5708
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.168627 0.180392 0.192157
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.15 1.4 0.1
      }
    }
  ]
  name "solid(3)"
}
DEF LineTrack Solid {
  translation -2.53001 -5.02 -0.049
  rotation 0 0 1 -3.1415853071795863
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.168627 0.180392 0.192157
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.15 1 0.1
      }
    }
  ]
  name "solid(4)"
}
LineTrack {
  translation -0.17 -4.52 0.0005
  rotation 0 0 1 1.5708
  scale 0.15 0.15 0.1
}
Robot {
  translation 0.150007 -2.68 0.0011
  children [
    Emitter {
      channel 1
      allowedChannels [
        1
      ]
    }
    GPS {
    }
    LineTrack {
      rotation 0 0 1 1.5708
      scale 0.15 0.15 0.0005
      color 1 0 0
    }
  ]
  name "ml"
  controller "possition_emitter_controller"
}
LineTrack {
  translation -0.009993 -4.52 0.0006
  rotation 0 0 1 1.5708
  scale 0.15 0.15 0.1
  color 1 0 0
}
Robot {
  translation 0.150005 2.49 0.0011
  children [
    Emitter {
      channel 1
      allowedChannels [
        1
      ]
    }
    LineTrack {
      rotation 0 0 1 1.5708
      scale 0.15 0.15 1
      color 1 0 0
    }
    GPS {
    }
  ]
  name "mr"
  controller "possition_emitter_controller"
}
Robot {
  translation -4.20999 2.515 0.0019
  children [
    Emitter {
      channel 1
      allowedChannels [
        1
      ]
    }
    LineTrack {
      rotation 0 0 1 1.5708
      scale 0.15 0.15 1
      color 1 0 0
    }
    GPS {
    }
  ]
  name "k"
  controller "possition_emitter_controller"
}
Robot {
  translation 0.150005 1.34 0.0028
  children [
    Emitter {
      channel 1
      allowedChannels [
        1
      ]
    }
    LineTrack {
      rotation 0 0 1 1.5708
      scale 0.15 0.15 1
      color 0 1 0
    }
    GPS {
    }
  ]
  name "mr_1"
  controller "possition_emitter_controller"
}
LineTrack {
  translation 0.320006 -4.52 0.0028
  rotation 0 0 1 1.5708
  scale 0.15 0.15 1
  color 0 1 0
}
LineTrack {
  translation 0.490006 -4.52 0.0028
  rotation 0 0 1 1.5708
  scale 0.15 0.15 1
  color 1 0.4 0.1
}
Robot {
  translation -0.749979 -2.675 0.0028
  children [
    Emitter {
      channel 1
    }
    LineTrack {
      rotation 0 0 1 1.5708
      scale 0.15 0.15 1
      color 1 0.4 0.1
    }
    GPS {
    }
  ]
  name "tbl_tl"
  controller "possition_emitter_controller"
}
Robot {
  translation -0.750003 2.49 0.0028
  children [
    Emitter {
      channel 1
    }
    LineTrack {
      rotation 0 0 1 1.5708
      scale 0.15 0.15 1
      color 1 0.4 0.1
    }
    GPS {
    }
  ]
  name "tbl_tr"
  controller "possition_emitter_controller"
}
Robot {
  translation -5.5 2.515 0.0028
  children [
    Emitter {
      channel 1
    }
    LineTrack {
      rotation 0 0 1 1.5708
      scale 0.15 0.15 1
      color 1 0.4 0.1
    }
    GPS {
    }
  ]
  name "k_in"
  controller "possition_emitter_controller"
}
Robot {
  translation 0.91 -2.675 0.0028
  children [
    Emitter {
      channel 1
    }
    LineTrack {
      rotation 0 0 1 1.5708
      scale 0.15 0.15 1
      color 1 0.4 0.1
    }
    GPS {
    }
  ]
  name "tbl_bl"
  controller "possition_emitter_controller"
}
Robot {
  translation 0.900001 2.49 0.0028
  children [
    Emitter {
      channel 1
    }
    LineTrack {
      rotation 0 0 1 1.5708
      scale 0.15 0.15 1
      color 1 0.4 0.1
    }
    GPS {
    }
  ]
  name "tbl_br"
  controller "possition_emitter_controller"
}
Robot {
  translation 2.51001 -0.55 0.0028
  children [
    Emitter {
      channel 1
    }
    LineTrack {
      rotation 0 0 1 1.5708
      scale 0.15 0.15 1
      color 1 0.4 0.1
    }
    GPS {
    }
  ]
  name "bl_2"
  controller "possition_emitter_controller"
}
Robot {
  translation 2.51001 0.4 0.0036
  children [
    Emitter {
      channel 1
    }
    LineTrack {
      rotation 0 0 1 1.5708
      scale 0.15 0.15 1
      color 1 0.4 0.1
    }
    GPS {
    }
  ]
  name "br_1"
  controller "possition_emitter_controller"
}
Robot {
  translation -2.50999 -0.55 0.0036
  children [
    Emitter {
      channel 1
    }
    LineTrack {
      rotation 0 0 1 1.5708
      scale 0.15 0.15 1
      color 1 0.4 0.1
    }
    GPS {
    }
  ]
  name "tl_2"
  controller "possition_emitter_controller"
}
Robot {
  translation -2.49999 0.39 0.0036
  children [
    Emitter {
      channel 1
    }
    LineTrack {
      rotation 0 0 1 1.5708
      scale 0.15 0.15 1
      color 1 0.4 0.1
    }
    GPS {
    }
  ]
  name "tr_1"
  controller "possition_emitter_controller"
}
Robot {
  translation -2.40999 4.44 0.0036
  children [
    Emitter {
      channel 1
    }
    LineTrack {
      rotation 0 0 1 1.5708
      scale 0.15 0.15 1
      color 0 0 1
    }
    GPS {
    }
  ]
  name "tr_2"
  controller "possition_emitter_controller"
}
Robot {
  translation -2.45999 -4.62 0.0036
  children [
    Emitter {
      channel 1
    }
    LineTrack {
      rotation 0 0 1 1.5708
      scale 0.15 0.15 1
      color 0 1 0
    }
    GPS {
    }
  ]
  name "tl_1"
  controller "possition_emitter_controller"
}
Robot {
  translation 2.77001 -4.6 0.0036
  children [
    Emitter {
      channel 1
    }
    LineTrack {
      rotation 0 0 1 1.5708
      scale 0.15 0.15 1
      color 0 1 0
    }
    GPS {
    }
  ]
  name "bl_1"
  controller "possition_emitter_controller"
}
Robot {
  translation 2.77001 4.45 0.0036
  children [
    Emitter {
      channel 1
    }
    LineTrack {
      rotation 0 0 1 1.5708
      scale 0.15 0.15 1
      color 0 0 1
    }
    GPS {
    }
  ]
  name "br_2"
  controller "possition_emitter_controller"
}
Robot {
  translation 0.150004 3.5 0.0028
  children [
    Emitter {
      channel 1
    }
    LineTrack {
      rotation 0 0 1 1.5708
      scale 0.15 0.15 1
      color 0 1 0
    }
    GPS {
    }
  ]
  name "mr_2"
  controller "possition_emitter_controller"
}
Robot {
  translation 0.160018 -1.5 0.0028
  children [
    Emitter {
      channel 1
      allowedChannels [
        1
      ]
    }
    LineTrack {
      rotation 0 0 1 1.5708
      scale 0.15 0.15 1
      color 0 0 1
    }
    GPS {
    }
  ]
  name "ml_2"
  controller "possition_emitter_controller"
}
Robot {
  translation 0.150018 -3.66 0.0028
  children [
    Emitter {
      channel 1
    }
    LineTrack {
      rotation 0 0 1 1.5708
      scale 0.15 0.15 1
      color 0 0 1
    }
    GPS {
    }
  ]
  name "ml_1"
  controller "possition_emitter_controller"
}
LineTrack {
  translation 0.150018 -4.52 0.0028
  rotation 0 0 1 1.5708
  scale 0.15 0.15 1
  color 0 0 1
}
Robot {
  translation -0.196354 -1.42932 0.699934
  rotation 0.010291209524042114 0.0008296333759598924 -0.9999466999370485 1.5228335463288776
  children [
    GPS {
      name "back_gps"
    }
    GPS {
      translation 0 0.1 0
      name "front_gps"
    }
    Emitter {
      translation -0.2 0 1.21
      children [
        DEF antena Solid {
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.01 0.01 0.25
              }
            }
          ]
          name "antena"
        }
      ]
      channel 1
      allowedChannels [
        1
      ]
    }
    Receiver {
      translation 0.2 0 1.21
      children [
        DEF antena Solid {
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.01 0.01 0.25
              }
            }
          ]
          name "antena"
        }
      ]
      channel 3
      allowedChannels [
        3
      ]
    }
    SpotLight {
      attenuation 0 0 1
      beamWidth 0.1
      intensity 0.5
      location 0 0.2 -0.58
    }
    Camera {
      translation 0 0.23 -0.67
      rotation 0 1 0 1.5699996938995748
      name "color_sensor"
    }
    DistanceSensor {
      translation 0.09 0.2 -0.67
      rotation 0 1 0 1.57
      name "ir_w_1"
      type "infra-red"
    }
    DistanceSensor {
      translation -0.09 0.2 -0.67
      rotation 0 1 0 1.57
      name "ir_w_2"
      type "infra-red"
    }
    Solid {
      translation -0.25 0.31 0.35
      children [
        Shape {
          geometry Box {
            size 0.1 0.6 0.1
          }
        }
      ]
      name "hand_2"
    }
    Solid {
      translation 0.25 0.31 0.35
      children [
        Shape {
          geometry Box {
            size 0.1 0.6 0.1
          }
        }
      ]
      name "hand_1"
    }
    Solid {
      translation 0 0.55 0.44
      children [
        Shape {
          appearance BlanketFabric {
          }
          geometry Box {
            size 0.6 0.7 0.1
          }
        }
      ]
      name "platter"
      boundingObject Box {
        size 0.6 0.7 0.1
      }
    }
    Solid {
      translation 0 0 0.89
      rotation 0 0 1 1.57
      children [
        Solid {
          translation 0.065 -0.085 0.03
          children [
            DEF eye Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.005 0.01 0.1
              }
            }
          ]
          name "r_eye"
        }
        Solid {
          translation 0.065 0.085 0.03
          children [
            DEF eye Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.005 0.01 0.1
              }
            }
          ]
          name "l_eye"
        }
        Solid {
          translation 0.01 0 0
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.7 0.8 1
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.1 0.4 0.3
              }
            }
          ]
          name "iner_screan"
        }
        Shape {
          appearance PBRAppearance {
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.1 0.5 0.4
          }
        }
      ]
      name "head"
    }
    DEF Body Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        roughness 1
        metalness 0
      }
      geometry Cylinder {
        height 1.4
        radius 0.2
      }
    }
    BallJoint {
      jointParameters BallJointParameters {
        position 3.2551577638375515
        anchor 0 0.5 -0.505
      }
      endPoint Solid {
        translation -0.018383777259005613 0.5082251671972029 -0.5061497609423501
        rotation -0.9094512653146949 -0.22774815725405356 0.34789247316503324 3.6908186798267226
        children [
          DEF ball_h Shape {
            appearance PBRAppearance {
              roughness 1
              metalness 0
            }
            geometry Sphere {
              radius 0.2
            }
          }
        ]
        name "solid(2)"
        boundingObject USE ball_h
        physics Physics {
          density 200
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 18.199184782688505
        anchor 0.25 0 -0.305
      }
      device [
        RotationalMotor {
          name "motor_1"
          maxTorque 100
        }
        PositionSensor {
          name "ps_1"
        }
      ]
      endPoint Solid {
        translation 0.2556438497863207 0.0045627259701821275 -0.30205191511791646
        rotation -0.6878963851553505 0.09339503740919235 -0.7197749163992431 2.964447849540747
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.1
              radius 0.4
            }
          }
        ]
        boundingObject USE Wheel
        physics DEF wheel_physics Physics {
          density 200
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 194.3355368518897
        anchor -0.25 0 -0.305
      }
      device [
        RotationalMotor {
          name "motor_2"
          maxTorque 100
        }
        PositionSensor {
          name "ps_2"
        }
      ]
      endPoint Solid {
        translation -0.26049582133073645 0.0043161083384444495 -0.3091963520805787
        rotation -0.677894007117365 0.16592786455293493 -0.7161896807964109 2.8168107086581924
        children [
          USE Wheel
        ]
        name "solid(1)"
        boundingObject USE Wheel
        physics USE wheel_physics
      }
    }
  ]
  name "waiter"
  boundingObject USE Body
  physics Physics {
    density 300
  }
  controller "waiter_controller"
}
Robot {
  translation -7.32 -4.79 0.49
  children [
    Emitter {
      translation 0.2 0.5 0.7
      children [
        DEF antena Solid {
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.02 0.02 0.5
              }
            }
          ]
          name "antena"
        }
      ]
      channel 3
      allowedChannels [
        5, 4, 3, 2
      ]
    }
    Receiver {
      translation 0.5 0.2 0.7
      children [
        DEF antena Solid {
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.02 0.02 0.5
              }
            }
          ]
          name "antena"
        }
      ]
      channel 1
      allowedChannels [
        1
      ]
    }
    Solid {
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.5 0.5 0.5
            roughness 1
            metalness 0
          }
          geometry Box {
            size 1 1 1
          }
        }
      ]
      name "body"
    }
  ]
  name "main_computer"
  controller "main_computer"
}
Wall {
  translation -7.93 0 0
  size 0.2 11 2.4
}
Wall {
  translation 5.12 0 0
  name "wall(1)"
  size 0.2 11 2.4
}
Wall {
  translation -1.42563 5.4 0
  rotation 0 0 1 1.57
  name "wall(2)"
  size 0.2 13 2.4
}
Wall {
  translation 2.13575 -5.39295 0
  rotation 0 0 1 1.57
  name "wall(3)"
  size 0.2 6 2.4
}
Wall {
  translation -5.94425 -5.38649 0
  rotation 0 0 1 1.57
  name "wall(4)"
  size 0.2 4 2.4
}
